docs/DeveloperGuide.md

```diff
@@ -749,49 +767,227 @@ _Customization of Contacts_
+8. Enter `find t/` in the command box.
+
+	Expected: Socialite shows error message, indicating tags to search for must not be empty.
+
+### Accessing social media page
+
+To test out this feature, you can choose to add your own contact information to SociaLite with your own social media handle. For example, you can enter `add n/<your name> p/<your phone number> ig/<your instagram handle>`.
+
+1. Hover your cursor on top of your Instagram handle in SociaLite.
+
+	Expected: the handle will turn underlined and the cursor will turn into a hand.
+
+2. Click your mouse when it is hovering on top of a certain handle.
+
+	Expected: SociaLite will open your default web browser and direct you to the specific social media page (e.g. your own Instagram profile).
+
+### Adding a profile picture to a contact
+
+Assuming there are 10 contacts stored in SociaLite:
+
+1. Enter `picture 1` in the command box.
+
+	Expected: a file chooser will show up for you to select your profile picture.
+
+	1. Click on `cancel` button or press `Esc` button in your OS.
+
+		Expected: you will be redirected to the SociaLite app, with error message indicating command aborted.
+
+	2. Locate a `.jpg` or `.png` file of your liking and select it in the file chooser.
+
+		Expected: the first contact card's profile picture will update to be the file that you have chosen.
+
+2. Enter `picture 100` in the command box.
+
+	Expected: SociaLite shows error message, indicating invalid person index provided.
+
+### Pinning/Unpinning a contact
+
+Assuming there are 10 contacts stored in SociaLite and none of them are pinned, try out the following test cases in sequence:
+
+1. Enter `pin 3` in the command box.
+
+	Expected: the third contact card in the list will be brought forward to the top of the list. The button "Pin" will change colour and its text will become "Unpin".
+
+2. Enter `unpin 1` in the command box.
+
+	Expected: the first contact card in the list (previously the third in the list) will be removed from the top of the list and go back to its original position. The button "Unpin" will change back to its original colour and its text will become "Pin".
+
+3. Enter `unpin 1` one more time in the command box.
+
+	Expected: SociaLite shows error message, indicating that the person is not pinned yet.
+
+4. Enter `pin 2` followed by `pin 1` in the command box.
+
+	Expected: Socialite shows error message, indicating that the person has already been pinned.
+
+5. Enter `pin 100` in the command box.
+
+	Expected: SociaLite shows error message, indicating invalid person index provided.
+
+6. Navigate to the sixth contact card in the GUI and click on the "Pin" button with your mouse.
+
+	Expected: The sixth contact card in the list will be pinned to the top of the list. The button "Pin" will change colour and its text will become "Unpin".
+
+7. Click on the "Unpin" button of the first contact card in the list with your mouse 
 
-### Launch and shutdown
+	Expected: same as test case number 2.
```

Comment:
Maybe it'll be better to elaborate on the expected outcome here because the outcome for Test Case 2 makes reference to the third contact in brackets. 

So we can say 'Expected: same as test case number 2. The chosen contact card will be unpinned and return to its original position in the list (i.e. in alphabetical order of names). The button "Unpin" will return to its original colour and its text will display "Pin".'.